src/game/client/draw/w3dmodeldraw.cpp

```diff
@@ -1154,3 +1178,1892 @@ void W3DModelDrawModuleData::Parse_Condition_State(INI *ini, void *instance, voi
+    m_isDaytime(true),
+    m_curState(nullptr),
+    m_hexColor(0),
+    m_renderObject(nullptr),
+    m_shadow(nullptr),
+    m_shadowState(true),
+    m_decalShadow(nullptr),
+    m_trackRenderObject(nullptr),
+    m_whichAnimInCurState(-1),
+    m_nextState(nullptr),
+    m_loopDuration(-1),
+    m_recalcBones(false),
+    m_fullyObscuredByShroud(false)
+{
+    for (int i = 0; i < WEAPONSLOT_COUNT; i++) {
+        m_weaponRecoilInfoVec[i].clear();
+    }
+
+    Get_W3D_Model_Draw_Module_Data()->Validate_Stuff_For_Time_And_Weather(Get_Drawable(),
+        g_theWriteableGlobalData->m_timeOfDay == TIME_OF_DAY_NIGHT,
+        g_theWriteableGlobalData->m_weather == WEATHER_SNOWY);
+
+    BitFlags<MODELCONDITION_COUNT> flags;
+    const ModelConditionInfo *idle = Find_Best_Info(flags);
+    captainslog_relassert(idle, CODE_06, "*** ASSET ERROR: all draw modules must have an IDLE state");
+
+    Drawable *drawable = Get_Drawable();
+
+    if (drawable != nullptr) {
+        Object *object = drawable->Get_Object();
+
+        if (object != nullptr) {
+            if (g_theWriteableGlobalData->m_timeOfDay == TIME_OF_DAY_NIGHT) {
+                m_hexColor = object->Get_Night_Indicator_Color();
+            } else {
+                m_hexColor = object->Get_Indicator_Color();
+            }
+        }
+
+        if (!Get_W3D_Model_Draw_Module_Data()->m_recievesDynamicLights) {
+            drawable->Set_Recieves_Dynamic_Lights(false);
+            captainslog_debug("setReceivesDynamicLights = FALSE: %s", drawable->Get_Template()->Get_Name().Str());
+        }
+    }
+
+    Set_Model_State(idle);
+}
+
+W3DModelDraw::~W3DModelDraw()
+{
+    if (m_trackRenderObject != nullptr && g_theTerrainTracksRenderObjClassSystem != nullptr) {
+        g_theTerrainTracksRenderObjClassSystem->Unbind_Track(m_trackRenderObject);
+        m_trackRenderObject = nullptr;
+    }
+
+    Nuke_Current_Render(nullptr);
+}
+
+void W3DModelDraw::On_Drawable_Bound_To_Object()
+{
+    Get_W3D_Model_Draw_Module_Data()->Validate_Stuff_For_Time_And_Weather(Get_Drawable(),
+        g_theWriteableGlobalData->m_timeOfDay == TIME_OF_DAY_NIGHT,
+        g_theWriteableGlobalData->m_weather == WEATHER_SNOWY);
+}
+
+void W3DModelDraw::Do_Start_Or_Stop_Particle_Sys()
+{
+    Drawable *drawable = Get_Drawable();
+    bool hidden = drawable->Is_Hidden() || m_fullyObscuredByShroud;
+
+    for (auto &tracker : m_particleSystemIDs) {
+        ParticleSystem *psys = g_theParticleSystemManager->Find_Particle_System(tracker.id);
+
+        if (psys != nullptr) {
+            if (hidden) {
+                psys->Stop();
+            } else {
+                psys->Start();
+            }
+        }
+    }
+}
+
+void W3DModelDraw::Set_Hidden(bool hidden)
+{
+    if (m_renderObject != nullptr) {
+        m_renderObject->Set_Hidden(hidden);
+    }
+
+    if (m_shadow != nullptr) {
+        m_shadow->Enable_Shadow_Render(!hidden);
+    }
+
+    m_shadowState = hidden;

```

Comment:
`m_isHidden` maybe?

Reply:
That makes sense and I can't find anything that suggests another name. Fixed.